cmd/fleet/main.go

```diff
@@ -783,7 +802,11 @@ func (f *FleetServer) runSubsystems(ctx context.Context, cfg *config.Config, g *
 	g.Go(loggedRunFunc(ctx, "Elasticsearch GC", sched.Run))
 
 	// Monitoring es client, longer timeout, no retries
-	monCli, err := es.NewClient(ctx, cfg, true, es.WithUserAgent(kUAFleetServer, f.bi))
+	options := []es.ConfigOption{es.WithUserAgent(kUAFleetServer, f.bi)}
+	if cfg.Inputs[0].Server.Instrumentation.Enabled {
+		options = append(options, es.InstrumentRoundTripper())
+	}
```

Comment:
nit: adding instrumentation to es client seems repeated in two places, maybe extract into a function?